samples/valueentity-counter-generated/src/main/java/com/example/domain/CounterProvider.java

```diff
@@ -0,0 +1,66 @@
+/* This code is managed by Akka Serverless tooling.
+ * It will be re-generated to reflect any changes to your protobuf definitions.
+ * DO NOT EDIT
+ */
+package com.example.domain;
+
+import com.akkaserverless.javasdk.valueentity.ValueEntityContext;
+import com.akkaserverless.javasdk.valueentity.ValueEntityOptions;
+import com.akkaserverless.javasdk.valueentity.ValueEntityProvider;
+import com.example.CounterApi;
+import com.example.domain.CounterDomain;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Empty;
+import com.google.protobuf.EmptyProto;
+import java.util.function.Function;
+
+/** A value entity provider */
+public class CounterProvider implements ValueEntityProvider {
```

Comment:
Why is this in `src/main/java`?

Reply:
what is `value-entity-counter-generated` compared to `value-entity-counter`? something we use for testing?

Reply:
Value entity counter has code that is generated using the maven plugin and we can't generate it when building with sbt. So, instead we do the following: 

* publish artifact locally
* generate code using the mvn plugin (cd samples/project && mvn compile) 
* the generated code is copied into `valueentity-counter-generated` in `src/main/java`. 
* the sbt project takes `value-entity-counter` and adds the code in `value-entity-counter-generated` as extra source. 

Now, when building with sbt, we can build `valueentity-counter` without needing the plugin. 
(you can find more info here: #171)

If it sounds complicated, is because it is. We have been trying a different things in an attempt to streamline the mixed build (sbt/mvn), but we didn't yet reach an optimal setup. 

The main advantage of this approach is that we can use IntelliJ to navigate through `valueentity-counter`, but the maintenance is high. I struggled today to get all things in place and got lost a few times.

Reply:
The ideal solution would be to have a sbt plugin for the codegen. The samples would be a first-class project in sbt and not the hacky solution we have right now. 

We will need such a plugin for the Scala SDK anyway. The plugin will need to be able to emit java as well (like the grpc plugin).

Reply:
Wow, that sounds indeed complicated. Nothing we need to change or improve now.

I haven't used that approach and didn't find it too difficult to try out new codegen in sample. I first used the ordinary tests `codegenJava/test` and then the instructions in https://github.com/lightbend/akkaserverless-java-sdk/blob/main/CONTRIBUTING.md

I compile/run the sample with mvn, and have an intellij maven project for the sample (have to modify the sdk version in pom to import project).